### ISSUE TITLE
problem on installation: iuio marked as required but not found

### ISSUE THREAD

**Issue by raymond00000** [VLOGroup/dvs-reconstruction#4]
Hi,

I followed the step to install imageutilities.
$ cd build
$ cmake ..
$ make
$ make install
(https://github.com/VLOGroup/imageutilities)

I set the IMAGEUTILITIES_ROOT to
/home/lscm/dvs/dvs-reconstruction/imageutilities

I installed cnpy,libcaer.

Then I ran below line and I got error message:
~/dvs/dvs-reconstruction/build$ cmake ../src
-- Imageutilities requested components: iucore;iuio;iugui;iumath
CMake Error at /home/lscm/dvs/dvs-reconstruction/imageutilities/ImageUtilitiesConfig.cmake:137 (message):
iuio marked as required but not found
Call Stack (most recent call first):
CMakeLists.txt:25 (find_package)

-- Configuring incomplete, errors occurred!

Thanks.

**Comment by raymond00000** [comment]
i guess i may missed the OpenCV2. let me try again and update soon.

**Comment by raymond00000** [comment]
i installed OpenCV, but I still got problem:
 
lscm@lscm-Dell-XPS420:~/dvs/dvs-reconstruction/imageutilities/build$ cmake ..
[-] ImageUtilities omitting IO module
[-] ImageUtilities omitting GUI module
[+] ImageUtilities include Math module
[-] do not build static library
Compiler Flags NVCC: -arch=sm_30-std=c++11
-- libs =ndarray
-- libs =ndarray
-- libs =ndarray
NDARRAY_LIB=ndarray
additional iuio libs: 
install: /home/lscm/dvs/dvs-reconstruction/imageutilities
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lscm/dvs/dvs-reconstruction/imageutilities/build



=========
/usr/local/lib$ ls
libcaer.so              libopencv_flann.a      libopencv_photo.a
libcaer.so.2            libopencv_gpu.a        libopencv_stitching.a
libcaer.so.2.0.0        libopencv_highgui.a    libopencv_superres.a
libcnpy.a               libopencv_imgproc.a    libopencv_ts.a
libcnpy.so              libopencv_legacy.a     libopencv_video.a
libopencv_calib3d.a     libopencv_ml.a         libopencv_videostab.a
libopencv_contrib.a     libopencv_nonfree.a    pkgconfig
libopencv_core.a        libopencv_objdetect.a  python2.7
libopencv_features2d.a  libopencv_ocl.a        python3.5

**Comment by reini1305** [comment]
You might have to delete the `CMakeCache.txt` file and run CMake again. Sometimes it tends to not find newly installed libraries.

**Comment by raymond00000** [comment]
acutally, i deleted the build folder and try again, but still,


~/dvs/dvs-reconstruction/imageutilities/build$ cmake ..
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found suitable version "8.0", minimum required is "5.0") 
[-] ImageUtilities omitting IO module
[-] ImageUtilities omitting GUI module
[+] ImageUtilities include Math module
[-] do not build static library
Compiler Flags NVCC: -arch=sm_30-std=c++11
-- libs =ndarray
-- libs =ndarray
-- libs =ndarray
NDARRAY_LIB=ndarray
additional iuio libs: 
install: /usr/local
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lscm/dvs/dvs-reconstruction/imageutilities/build


is that any flag needed to build with IO module? thank a lot.

**Comment by reini1305** [comment]
You only seem to have built the static version of OpenCV. There are no .so files...

**Comment by raymond00000** [comment]
i ran this sh
https://gist.github.com/arthurbeggs/06df46af94af7f261513934e56103b30

**Comment by raymond00000** [comment]
let me try apt-get version..and update soon.
sudo apt-get install libopencv-dev python-opencv
thanks for help.

**Comment by reini1305** [comment]
line 21: `-DBUILD_SHARED_LIBS=OFF` should be ON.

**Comment by raymond00000** [comment]
thank a lot, let me try again.

**Comment by raymond00000** [comment]
I tried again, now I got:

lscm@lscm-Dell-XPS420:/usr/local/lib$ ls
libcaer.so                      libopencv_ml.a
libcaer.so.2                    libopencv_ml.so
libcaer.so.2.0.0                libopencv_ml.so.2.4
libcnpy.a                       libopencv_ml.so.2.4.13
libcnpy.so                      libopencv_nonfree.a
libopencv_calib3d.a             libopencv_nonfree.so
libopencv_calib3d.so            libopencv_nonfree.so.2.4
libopencv_calib3d.so.2.4        libopencv_nonfree.so.2.4.13
libopencv_calib3d.so.2.4.13     libopencv_objdetect.a
libopencv_contrib.a             libopencv_objdetect.so
libopencv_contrib.so            libopencv_objdetect.so.2.4
libopencv_contrib.so.2.4        libopencv_objdetect.so.2.4.13
libopencv_contrib.so.2.4.13     libopencv_ocl.a
libopencv_core.a                libopencv_ocl.so
libopencv_core.so               libopencv_ocl.so.2.4
libopencv_core.so.2.4           libopencv_ocl.so.2.4.13
libopencv_core.so.2.4.13        libopencv_photo.a
libopencv_features2d.a          libopencv_photo.so
libopencv_features2d.so         libopencv_photo.so.2.4
libopencv_features2d.so.2.4     libopencv_photo.so.2.4.13
libopencv_features2d.so.2.4.13  libopencv_stitching.a
libopencv_flann.a               libopencv_stitching.so
libopencv_flann.so              libopencv_stitching.so.2.4
libopencv_flann.so.2.4          libopencv_stitching.so.2.4.13
libopencv_flann.so.2.4.13       libopencv_superres.a
libopencv_gpu.a                 libopencv_superres.so
libopencv_gpu.so                libopencv_superres.so.2.4
libopencv_gpu.so.2.4            libopencv_superres.so.2.4.13
libopencv_gpu.so.2.4.13         libopencv_ts.a
libopencv_highgui.a             libopencv_video.a
libopencv_highgui.so            libopencv_video.so
libopencv_highgui.so.2.4        libopencv_video.so.2.4
libopencv_highgui.so.2.4.13     libopencv_video.so.2.4.13
libopencv_imgproc.a             libopencv_videostab.a
libopencv_imgproc.so            libopencv_videostab.so
libopencv_imgproc.so.2.4        libopencv_videostab.so.2.4
libopencv_imgproc.so.2.4.13     libopencv_videostab.so.2.4.13
libopencv_legacy.a              pkgconfig
libopencv_legacy.so             python2.7
libopencv_legacy.so.2.4         python3.5
libopencv_legacy.so.2.4.13


Then I rebuilt imageultiities:

lscm@lscm-Dell-XPS420:~/dvs/dvs-reconstruction/imageutilities$ rm -rf build/
lscm@lscm-Dell-XPS420:~/dvs/dvs-reconstruction/imageutilities$ mkdir build
lscm@lscm-Dell-XPS420:~/dvs/dvs-reconstruction/imageutilities$ cd build/
lscm@lscm-Dell-XPS420:~/dvs/dvs-reconstruction/imageutilities/build$ cmake ..
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found suitable version "8.0", minimum required is "5.0") 
[-] ImageUtilities omitting IO module
[-] ImageUtilities omitting GUI module
[+] ImageUtilities include Math module
[-] do not build static library
Compiler Flags NVCC: -arch=sm_30-std=c++11
-- libs =ndarray
-- libs =ndarray
-- libs =ndarray
NDARRAY_LIB=ndarray
additional iuio libs: 
install: /usr/local
-- Configuring done
-- Generating done
-- Build files have been written to: /home/lscm/dvs/dvs-reconstruction/imageutilities/build


But still, i see this:
[-] ImageUtilities omitting IO module
[-] ImageUtilities omitting GUI module

no idea on what is wrong...



then i ran build for dvs-reconstruction:

lscm@lscm-Dell-XPS420:~/dvs/dvs-reconstruction/build$ cmake ../src
-- The C compiler identification is GNU 5.4.0
-- The CXX compiler identification is GNU 5.4.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found suitable version "8.0", minimum required is "5.0") 
-- Imageutilities requested components: iucore;iuio;iugui;iumath
CMake Error at /home/lscm/dvs/dvs-reconstruction/imageutilities/ImageUtilitiesConfig.cmake:137 (message):
  iuio marked as required but not found
Call Stack (most recent call first):
  CMakeLists.txt:25 (find_package)


-- Configuring incomplete, errors occurred!
See also "/home/lscm/dvs/dvs-reconstruction/build/CMakeFiles/CMakeOutput.log".
See also "/home/lscm/dvs/dvs-reconstruction/build/CMakeFiles/CMakeError.log".
